datastore/datastore.go

```diff
@@ -37,23 +50,35 @@ type KV interface {
 const (
 	// NetworkKeyPrefix is the prefix for network key in the kv store
 	NetworkKeyPrefix = "network"
+	// EndpointKeyPrefix is the prefix for endpoint key in the kv store
+	EndpointKeyPrefix = "endpoint"
 )
 
+var rootChain = []string{"docker", "libnetwork"}
+
 //Key provides convenient method to create a Key
 func Key(key ...string) string {
-	keychain := []string{"docker", "libnetwork"}
-	keychain = append(keychain, key...)
+	keychain := append(rootChain, key...)
 	str := strings.Join(keychain, "/")
 	return str + "/"
 }
 
+//UnKey provides convenient method to unpack the key to compliment the Key function
+func UnKey(key string) ([]string, error) {
```

Comment:
I don't think the name `Unkey` is intuitive to let us know what operation is being done. A more appropriate name would be `ParseKey`